### ISSUE TITLE
Replace die.net links with better resources where applicable

### ISSUE THREAD

**Issue by mebeim** [tldr-pages/tldr#5143]
There was [a short discussion](https://gitter.im/tldr-pages/tldr?at=600008e514cec811ec998765) today on Gitter about pages which have "more information" links pointing to `die.net`. As @navarroaxel showed there is man.archlinux.org, which is for Arch Linux programs, and which could be better suited for Arch Linux specific utilities.

In general though, we have the same distro-generic set of manual pages that `die.net` offers on `man7.org`, which IMHO renders the pages a lot better (sometimes `die.net` pages have syntax errors due to special characters being treated strangely when rendered in HTML, furthermore `man7.org` pages are actually in monospace). 

Therefore I suggest we convert all "more information" links which are pointing to `die.net` to point to `man7.org` instead, and in case the package/command is distro-specific, directly point to the distribution's own manual pages if available, for example:

- [Ubuntu](http://manpages.ubuntu.com/manpages/)
- [Debian](https://manpages.debian.org/buster/man-db/index.html)
- [Arch](https://man.archlinux.org)
- etc...

**Comment by sbrl** [comment]
Sounds like a good idea!

**Comment by bl-ue** [comment]
How about https://man7.org? It's seems distro-agnostic (at least the url is), and it's updated very frequently.  Two days ago as of 1/31/2021.

Once we decide, I'll go through and fix everything.

**Comment by bl-ue** [comment]
Oops sorry I barely even read your message @mebeim! man7.org it is.

**Comment by bl-ue** [comment]
I'll tackle this now.

**Comment by sbrl** [comment]
Sounds good to me. Although, should a command be distribution specific (e.g. `dpkg`, or `apt`) we should reference the source distribution (in the case of that example, it would be the Debian website I'd assume).

**Comment by marchersimon** [comment]
Oh, sorry I didn't see you were working on this.

**Comment by bl-ue** [comment]
Don't be sorry—I'm glad! I have a lot of things I've assigned myself and I'm glad an energetic helper has shouldered one of the burdens! 🙂

**Comment by vladimyr** [comment]
> As @navarroaxel showed there is man.archlinux.org, which is for Arch Linux programs, and which could be better suited for Arch Linux specific utilities.

Arch manpages are not exclusively used for Arch utilities. Take [`vmstat(8)`](https://man.archlinux.org/man/vmstat.8) for example. It is part of an independent project (`procps`) living at https://gitlab.com/procps-ng/procps. In other words, Arch folks host distro-generic stuff too.

> In general though, we have the same distro-generic set of manual pages that `die.net` offers on `man7.org`, which IMHO renders the pages a lot better (sometimes `die.net` pages have syntax errors due to special characters being treated strangely when rendered in HTML, furthermore `man7.org` pages are actually in monospace).

Oh, you are concerned about quirky HTML conversions and rendering, I feel your pain. Let me show you something really nice: https://gitlab.archlinux.org/archlinux/archmanweb/-/blob/275f3ee173ae658a6eedc139c49002371192728a/archmanweb/urls.py#L12
What that means is that you can switch between different formats:
- regular HTML output - https://man.archlinux.org/man/vmstat.8.html (OFC `.html` bit is optional)
- plaintext output - https://man.archlinux.org/man/vmstat.8.txt
- manpage source - https://man.archlinux.org/man/vmstat.8.raw

Also take, a look at the actual HTML source code of that `vmstat(8)` page, no JavaScript at all! OTOH, while I do appreciate Michael's work on man7.org it is a bit uncool to stick two (controversial) trackers into someone's face:
```html
<!--BEGIN-SITETRACKING-->
<!-- SITETRACKING.man7.org_linux_man-pages -->

<!-- Default Statcounter code for man7.org/linux/man-pages
http://www.man7.org/linux/man-pages -->
<script type="text/javascript">
var sc_project=7422636;
var sc_invisible=1;
var sc_security="9b6714ff";
</script>
<script type="text/javascript"
src="https://www.statcounter.com/counter/counter.js"
async></script>
<noscript><div class="statcounter"><a title="Web Analytics
Made Easy - StatCounter" href="https://statcounter.com/"
target="_blank"><img class="statcounter"
src="https://c.statcounter.com/7422636/0/9b6714ff/1/"
alt="Web Analytics Made Easy -
StatCounter"></a></div></noscript>
<!-- End of Statcounter Code -->



<!-- Start of Google Analytics Code -->

<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-9830363-8']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- End of Google Analytics Code -->

<!--END-SITETRACKING-->
```
taken from https://man7.org/linux/man-pages/man8/vmstat.8.html 

> Therefore I suggest we convert all "more information" links which are pointing to `die.net` to point to `man7.org` instead, and in case the package/command is distro-specific, directly point to the distribution's own manual pages if available, for example:
> 
> * [Ubuntu](http://manpages.ubuntu.com/manpages/)
> * [Debian](https://manpages.debian.org/buster/man-db/index.html)
> * [Arch](https://man.archlinux.org)
> * etc...

I propose Arch manpages as default, man7.org as a fallback (in case of page not being present in Arch manpage set), and using distro-specific man frontends where needed.

**Comment by vladimyr** [comment]
For the record here is a complete (check)list of pages pointing to `linux.die.net`:
- [ ] addpart
- [ ] aplay
- [ ] arecord
- [ ] cfdisk
- [ ] dumpe2fs
- [ ] e2freefrag
- [ ] e2image
- [ ] efibootmgr
- [ ] expect
- [ ] filefrag
- [ ] ifdown
- [ ] iftop
- [ ] ifup
- [ ] iotop
- [ ] isosize
- [ ] iwconfig
- [ ] jpegtran
- [ ] lftp
- [ ] logsave
- [ ] ltrace
- [ ] mdadm
- [ ] named
- [ ] pdflatex
- [ ] pmount
- [ ] semanage
- [ ] snmpwalk
- [ ] sponge
- [ ] transmission-create
- [ ] vipw
- [ ] vmstat
- [ ] xar

Generated using `ripgrep`:
```sh
rg -l 'linux.die.net' | xargs -n1 basename -s '.md' | sort | xargs -n1 printf '- [ ] %s\n'
```

**Comment by vladimyr** [comment]
I wrote a quick and dirty script to check for manpages on Arch end:

<details>
<summary><code>check-manpages</code></summary>

```bash
#!/bin/bash
set -euo pipefail

pushd "$HOME/.tldr/cache/pages" >/dev/null 2>&1

pages=$(rg -o 'linux.die.net/man/([^/])/([^>]+)' -r '$2.$1' | sort)

for page in $pages; do
  tldrpage=$(echo "$page" | cut -d':' -f1)
  manpage=$(echo "$page" | cut -d':' -f2)
  url="https://man.archlinux.org/man/$manpage"
  status=$(curl -sI "$url" | grep 'HTTP/')
  printf '%s:\t%s\t%s\n' "$tldrpage" "$manpage" "$status"
done

popd >/dev/null 2>&1

```
</details> 

And here is what I got:
```
❯ bash check-manpages | grep 404
common/iotop.md:        iotop.1 HTTP/2 404 
linux/ifdown.md:        ifdown.8        HTTP/2 404 
linux/ifup.md:  ifup.8  HTTP/2 404 
linux/pmount.md:        pmount.1        HTTP/2 404 
linux/semanage.md:      semanage.8      HTTP/2 404 
osx/xar.md:     xar.1   HTTP/2 404 
```

`xar` is obviously an outlier here because it is macOS specific and for the rest, it seems that only `semanage(8)` is available on man7.org (checked using [`inurl:"iotop.1" OR inurl:"ifdown.8" OR inurl:"ifup.8" OR inurl:"pmount.1" OR inurl:"semanage.8" site:man7.org/linux/man-pages`](https://www.google.com/search?q=inurl%3A%22iotop.1%22+OR+inurl%3A%22ifdown.8%22+OR+inurl%3A%22ifup.8%22+OR+inurl%3A%22pmount.1%22+OR+inurl%3A%22semanage.8%22+site%3Aman7.org%2Flinux%2Fman-pages&ei=jDhhYIOLDsWAkwWEwrCoCQ&oq=inurl%3A%22iotop.1%22+OR+inurl%3A%22ifdown.8%22+OR+inurl%3A%22ifup.8%22+OR+inurl%3A%22pmount.1%22+OR+inurl%3A%22semanage.8%22+site%3Aman7.org%2Flinux%2Fman-pages)).

**Comment by vladimyr** [comment]
`ifup` and `ifdown` are part of `ifupdown` suite maintained by Debian:
- https://manpages.debian.org/latest/ifupdown/ifup.8
- https://manpages.debian.org/latest/ifupdown/ifdown.8

`pmount` is also maintained by Debian folks:
- https://manpages.debian.org/latest/pmount/pmount.1

`iotop` is an independent project - http://guichaz.free.fr/iotop/ but the author in the [manpage](https://repo.or.cz/iotop.git/blob/1bfb3bc70febb1ffb95146b6dcd65257228099a3:/iotop.8#l98) itself says it is made (manpage, not `iotop`) for Debian:
- https://manpages.debian.org/latest/iotop/iotop.8
Note that die.net places this manpage in the wrong section https://linux.die.net/man/1/iotop :upside_down_face: 

I'm not aware of any RHEL or macOS specific online manpage viewers :shrug:

**Comment by marchersimon** [comment]
@vladimyr I now replaced all links with the arch man page where possible in my PR (#5528). Should I search alternatives for the other `man7.org` pages too?

**Comment by vladimyr** [comment]
@marchersimon Well, I like that you appreciate my opinion on the subject but let's wait for others to have their say and reach a consensus.

**Comment by waldyrious** [comment]
> I now replaced all links with the arch man page where possible in my PR (#5528).

I'd rather we reach explicit consensus here before making the changes. I'd suggest holding off that PR for now, until we reach a decision.

@mebeim, @sbrl — you two expressed preference towards man7.org; any thoughts on @vladimyr's comments?

**Comment by mebeim** [comment]
Thanks for the ping @waldyrious. 

@vladimyr I see your point and I must agree, it seems that Arch man pages are a way better alternative to `man7`. I'd suggest pointing to those if they provide arch-agnostic pages, or to the appropriate distro (e.g. Debian/Ubuntu/etc) if the tool is distro specific.

**Comment by waldyrious** [comment]
@mebeim do you also think we should prefer Ubuntu manpages over unix.com? I'm trying to understand the motivation for [this comment](https://github.com/tldr-pages/tldr/pull/5528#discussion_r604477703).

**Comment by vladimyr** [comment]
Folks, just a quick heads up, I have made some recent discoveries that actually made me alter my proposal 🙃 
I'm AFK at the moment but expect a proper update on this in a few hours...

**Comment by vladimyr** [comment]
Ok, I'm back...

First I need to clear things up. I wasn't precise enough stating:

> In other words, **Arch folks host distro-generic stuff** too.

They do host stuff that is not necessarily Arch specific but only if it is part of the core Arch package set or distributed through AUR so there is some truth in @mebeim's earlier statement:

> As @navarroaxel showed there is man.archlinux.org, which is **for Arch Linux programs**, and which could be better suited for Arch Linux specific utilities.

What it means is that we still need to fallback to other manpage frontends in cases like the ones I described here: https://github.com/tldr-pages/tldr/issues/5143#issuecomment-809037244 @mebeim summarized it well:

> I'd suggest pointing to those if they provide arch-agnostic pages, or to the appropriate distro (e.g. Debian/Ubuntu/etc) if the tool is distro specific.

Sure that works and it is definitely an upgrade compared to linux.die.net, and a better alternative than man7.org IMHO. However, I still wished for something that will cover all our needs without having to resort to alternatives. Basically, something like https://www.unix.com/man-page-repository.php, covering many different manpage sources and actually user-friendly. (I think it goes without saying but unix.com is out due to horrendous formatting and page bloat, ranging from visual clutter straight up to ads.)

And I actually found it by reading archman's README: https://gitlab.archlinux.org/archlinux/archmanweb#similar-projects 
Meet https://manned.org :tada: It is quite an old project developed by `ncdu` author: https://dev.yorhel.nl/ncdu and pretty much ticks all boxes:
- Albeit not very visually appealing it renders manpages nicely using proper monospace font: https://manned.org/ls.1
- It gives you the ability to pick a different section or language related to the current manpage.
- It provides the ability to choose between different manpage versions or manpage from a different distro.
- It has a permalink option together with the source url.
- It does use Javascript but **not** for page rendering so users who disable Javascript on the web still can view the actual content.
- It is opensource and actively maintained: https://code.blicky.net/yorhel/manned/commits/branch/master
- It supports a broad set of distros: https://code.blicky.net/yorhel/manned/src/branch/master/sql/schema.sql#L54-L258
- It is regularly updated, has long operating history and it is nicely documented: https://manned.org/info/about 
- It uses well-thought search algorithm used for determining the preferred page: https://code.blicky.net/yorhel/manned/src/branch/master/www/index.pl#L1008-L1017

It is important to note that manned.org search algorithm is tuned in a such way that in most cases it actually gives you the same content as archman would which is a great thing as AUR is both large and regularly updated. In cases given manpage does not belong in Arch manpage set it will transparently switch to appropriate fallback which essentially means that we don't need to do it manually. 
Take `pmount` for example https://manned.org/pmount.8 It somehow managed to beat archman in its own court. Here is what archman gives you: https://man.archlinux.com/pmount.8 i.e. nothing although it is clearly distributed through AUR: https://aur.archlinux.org/packages/pmount OTOH manned.org found it there and gives you manpage straight from the latest published version :rocket: 
Let's try RHEL's `semanage`: https://manned.org/semanage.8 voila, straight from the CentOS manpage set. Or macOS specific `xar`: https://manned.org/xar.1 which serves us better than fairly older https://www.unix.com/man-page/osx/1/xar/

It is so good it feels almost like magic!

So, to summarise, my _new_ proposal is: let's use manned.org everywhere! :tada:

**Comment by bl-ue** [comment]
Wow @vladimyr that's a great find! The formatting is nice and simple, and sure enough I found `ip link` with all of it's full documentation and even examples at the bottom: https://manned.org/ip-link

Also this is what they say on the homepage:

>Indexing **5,502,148** versions of **335,112** manual pages found in **21,062,967** files of **1,414,023** packages.

Unfortunately, we need to use some other source for Gentoo pages because as they source towards the bottom of the [about page](https://manned.org/info/about) (bold mine):

>Suggestions for new (or old) systems to index are welcome.

>It would be great to index a few more non-Linux systems such as other BSDs, Solaris/Illumos and Mac OS X. Unfortunately, those don't always follow a binary package based approach, or are otherwise less easy to properly index.

>In general, systems that follow an entirely source-based distribution approach can't be indexed without compiling everything. Since that is both very resource-heavy and open to security issues, there are no plans to include manuals from such systems at the moment. **So unless someone comes with a solution I hadn't thought of yet, there won't be any Gentoo manuals here. :-(**

**Comment by waldyrious** [comment]
Do we even _have_ any Gentoo-only pages?

**Comment by bl-ue** [comment]
In fact, I have no idea! 😄 I was just giving a heads up.

**Comment by vladimyr** [comment]
> In fact, I have no idea! 😄 I was just giving a heads up.

Which I fully appreciate as I'm known as someone who doesn't read till the end of readme like in archman's case 🙃

With that being said I'm not aware of any online viewer for Gen/Funtoo manpages? I tried searching for the `emerge` manpage online and this is all I got https://dev.gentoo.org/~zmedico/portage/doc/man/emerge.1.html It's basically an output of man2html hosted by Gentoo dev on his personal space and that only covers portage. Are there better sources out there?

**Comment by sbrl** [comment]
Great discussion here. I'm not sure it's as clear-cut though as saying "let's use the Arch Linux man website for everything", because some commands are distro-specific. For example, `apt` is a Debian command, and thus we should link to the Debian man page because it's unlikely to be found in other distros. `dnf` is a Fedora / red hat command, so we should like to the disstro-specific docs. `emerge` is a _Gentoo_ command, so we should link to the Gentoo docs.

This said, there are also plenty of commands that are distro-agnostic. Picking a good (and _stable_ / _long lasting_) website here is a good idea.

I do agree that the Arch Linux help pages are very helpful, but I'm unsure about blindly linking to them as a catch-all here, because some man pages there might be distro-specific. So I'm inclined to choose a more neutral contender to solve the issue here.

I especially like the suggestion to reference <https://manned.org> here as our catch-all. As mentioned by @vladimyr above, it seems to be well-maintained - and [according to the web archive](http://web.archive.org/web/20121111053053/http://manned.org/) it has been around in it's present form since 2012 (before which it was a website about manned space flight, which started in 2006 and ended in 2010).

Of course, for distro-specific man pages I say that we reference the distro's website, but for a catch-all <https://manned.org/> is the best thing I've seen (it even has a DuckDuckGo bang: `!manned`).
